### PR TITLE
optimize CollectionAddonViewset with transforms and attach_trans_dict

### LINKED ISSUE #16081: optimize CollectionAddonViewSet queryset performance
follow on from #8831 (that added caching).

A single list request for CollectionAddonViewSet makes a large number of queries, and some of them are per addon, so it doesn't scale well.  We should change the queries so that at least all the results are retrieved in a single query - a collection with 100 addons should have the same number of queries as a collection with 1 addon inside.

### PR DESCRIPTION
**Pull request by eviljeff**
fixes #16081

### CODE REVIEW

**Review by bobsilverberg — APPROVED** [review]
I can't say that I really understand what's going on in some of these files, but nothing jumps out at me as a problem, and I see that Mat has also been asked to review.

File: src/olympia/bandwagon/views.py [hunk 1]
```diff
@@ -138,11 +139,38 @@ def get_object(self):
...
+        attach_trans_dict(CollectionAddon, objs)
+        attach_trans_dict(Addon, addons)
+        attach_trans_dict(License, [ver.license for ver in current_versions])
+
     def get_queryset(self):
         qs = (
             CollectionAddon.objects
             .filter(collection=self.get_collection())
-            .prefetch_related('addon'))
+            .prefetch_related('addon__promotedaddon', 'addon__addon_tags')
```
Comment (diox) [inline src/olympia/bandwagon/views.py:168]: Why do you need `addon__addon_tags` here, shouldn't it be handled by your call to `attach_tags` in the `_transformer` ?

**Review by diox — COMMENTED** [review]
Needs some tests to ensure the transformer are fetching the right data and setting it correctly

File: src/olympia/bandwagon/views.py (hunk 1 above)
Comment (eviljeff) [reply src/olympia/bandwagon/views.py]: you're right, that's one more query down https://github.com/mozilla/addons-server/pull/16082/commits/aa96ec8673353d9bc86611d02f2c5679167cf3fe

**Comment by eviljeff** [comment]
Added a test that checks the localisations were loaded correctly https://github.com/mozilla/addons-server/pull/16082/commits/aa96ec8673353d9bc86611d02f2c5679167cf3fe

**Review by diox — APPROVED** [review]
:1st_place_medal:

**Pull request merged**